python_src/src/lamp_py/performance_manager/flat_file.py

```diff
@@ -0,0 +1,130 @@
+)
+from lamp_py.postgres.postgres_utils import DatabaseManager
+
+
+def generate_daily_table(
+    db_manager: DatabaseManager, service_date: int
+) -> pyarrow.Table:
+    """
+    Generate a dataframe of all events and metrics for a single service date
+    """
+    query = (
+        sa.select(
+            VehicleEvents.stop_sequence,
+            VehicleEvents.stop_id,
+            VehicleEvents.parent_station,
+            VehicleEvents.vp_move_timestamp.label("move_timestamp"),
+            sa.func.coalesce(
+                VehicleEvents.vp_stop_timestamp,
+                VehicleEvents.tu_stop_timestamp,
+            ).label("stop_timestamp"),
+            VehicleEvents.travel_time_seconds,
+            VehicleEvents.dwell_time_seconds,
+            VehicleEvents.headway_trunk_seconds,
+            VehicleEvents.headway_branch_seconds,
+            VehicleTrips.service_date,
+            VehicleTrips.route_id,
+            VehicleTrips.direction_id,
+            VehicleTrips.start_time,
+            VehicleTrips.vehicle_id,
+            VehicleTrips.branch_route_id,
+            VehicleTrips.trunk_route_id,
+            VehicleTrips.stop_count,
+            VehicleTrips.trip_id,
+            VehicleTrips.vehicle_label,
+            VehicleTrips.vehicle_consist,
+            VehicleTrips.direction,
+            VehicleTrips.direction_destination,
+            VehicleTrips.static_start_time,
+            StaticStopTimes.arrival_time.label("scheduled_arrival_time"),
+            StaticStopTimes.departure_time.label("scheduled_departure_time"),
+            StaticStopTimes.schedule_travel_time_seconds.label(
+                "scheduled_travel_time"
+            ),
+            StaticStopTimes.schedule_headway_branch_seconds.label(
+                "scheduled_headway_branch"
+            ),
+            StaticStopTimes.schedule_headway_trunk_seconds.label(
+                "scheduled_headway_trunk"
+            ),
+        )
+        .join(VehicleTrips, VehicleEvents.pm_trip_id == VehicleTrips.pm_trip_id)
+        .join(
+            StaticStopTimes,
+            sa.and_(
+                StaticStopTimes.static_version_key
+                == VehicleTrips.static_version_key,
+                StaticStopTimes.trip_id == VehicleTrips.static_trip_id_guess,
+                StaticStopTimes.stop_id == VehicleEvents.stop_id,
+            ),
+            isouter=True,
+        )
+        .where(
+            sa.and_(
+                VehicleEvents.service_date == service_date,
+                sa.or_(
+                    VehicleEvents.vp_move_timestamp.is_not(None),
+                    VehicleEvents.vp_stop_timestamp.is_not(None),
+                ),
+            )
+        )
+    )
+
+    # get the days events as a dataframe from postgres
+    days_events = db_manager.select_as_dataframe(query)
+
+    # transform the seru
+    days_events["year"] = days_events.apply(
+        lambda record: int(str(record["service_date"])[0:4]), axis=1
```

Comment:
this can be down without apply/lambda

`days_events["year"] = days_events["service_date"].astype(str).str[:4].astype(int)`

performance diff:
astype: 356 µs ± 9.46 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
apply: 2.16 ms ± 11.8 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)